### ISSUE TITLE
Add CD for npm publish

### ISSUE THREAD

**Issue by lukasjhan** [openwallet-foundation-labs/sd-jwt-js#44]
## Overview

OWF provide npm token for our repo. you can check it in discord channel.
We'll apply same way with https://github.com/openwallet-foundation/agent-framework-javascript/blob/main/.github/workflows/continuous-deployment.yml

But, I think it should be triggered by release

**Comment by cre8** [comment]
One good approach I saw:
- the code on main will always be published under the tag next
- versions will be managed by creating a new release
- release documents are auto generated by semantic release commits lik `fix: ...`

**Comment by lukasjhan** [comment]
That approach seems good. Currently I deploy it manually but OWF provide us a npm token, so next time, let's use it to publish automatically

I think triggering the release to the npm when we make a new release in github is better. :)
What do you think? @cre8

**Comment by cre8** [comment]
Before choosing a way, I would look at the used framework that should be used to manage a monorepo when we want to split it up.

E.g. Lerna has a lot of functions already for versioning and i would prefer to reuse these common approaches so we can focus to work on the lib and not the "managing stuff" around it.

**Comment by lukasjhan** [comment]
I'll make a PR for integration with Lerna.

**Comment by lukasjhan** [comment]
@berendsliedrecht @cre8 

Based #68, now we can discuss the package names. I believe @openwalletfounction/* will be used since there is a token provided by OWF. 

I made name in this format: sd-jwt-{sub package name}. so e.g. sd-jwt-type, sd-jwt-core.
I would appreciate it if you could share your thoughts please :)

**Comment by cre8** [comment]
the prefix with openwalletfoundation is fine for me, since owf is already taken.

also the approach with the general prefix `sd-jwt` is fine for me.

To make the versioning easier I would suggest to sync the version numbers over the packages. In this case when we release something, we have it like:
```
Version 1.1.0
sd-jwt-core:
- fixed bug foo bar
```

**Comment by lukasjhan** [comment]
Version sync is great. If not, there will be a compatible issue between each versions

**Comment by lukasjhan** [comment]
I guess there is no package named @owf/*.
How can I check owf is used or not? @cre8

**Comment by cre8** [comment]
Any chances we can reuse the @sd-jwt that is used right now?

Other typescript projects from the owf like credo also use their own prefix.

**Comment by berendsliedrecht** [comment]
We can use the `@sd-jwt`, yes! Also, installing, or importing, multiple packages with the `@openetfoundation` can get quite annoying as it is fairly long :).

**Comment by lukasjhan** [comment]
I like @sdj-wt too. :) 

0.2.1 is the latest version of @sd-jwt packages

**Comment by lukasjhan** [comment]
That sounds good to me.

**Comment by cre8** [comment]
okay, I can look at the versioning stuff with lerna

**Comment by cre8** [comment]
in case we want to publish to npmjs, yes, see https://docs.github.com/de/actions/publishing-packages/publishing-nodejs-packages

**Comment by lukasjhan** [comment]
@berendsliedrecht  Can you prepare an npm token so I can deploy this repo to @sd-jwt please?

**Comment by berendsliedrecht** [comment]
> @berendsliedrecht  Can you prepare an npm token so I can deploy this repo to @sd-jwt please?

Yes! Just trying to think now of the best way to set this. @ryjones any tips? Do  I just need to supply you the token and can you add the secret?

**Comment by cre8** [comment]
Someone with admin privileges of this repo has to set the the token created by you @berendsliedrecht and store it as a secret in this repo. We can configure that it can only be used when the cicd runs on a protected branch.

**Comment by ryjones** [comment]
please make me an owner over on npm and I'll create and add the token.

**Comment by lukasjhan** [comment]
@berendsliedrecht Could you invite me too? :) 
`lukas.j.han@gmail.com`

**Comment by lukasjhan** [comment]
I'll change the name in package.json today :)

**Comment by lukasjhan** [comment]
done in #77

**Comment by cre8** [comment]
@lukasjhan will you make a PR or should I?

**Comment by ryjones** [comment]
@lukasjhan @cre8 the secret is named `NPM_TOKEN`. Do I need to rename it?

**Comment by cre8** [comment]
@ryjones @lukasjhan we do not need to rename it, I know why the ci is falling. It's because of this:
          token: ${{secrets.GH_TOKEN}}

I fixed it on one of my branches, but didn't make a PR. So we do not need to update the secret name, but fix the workflow (I will do it tomorrow)

**Comment by lukasjhan** [comment]
Okay :) It was my misunderstanding
